compiler/parser/parser.cpp

```diff
@@ -793,6 +801,57 @@ static void addModuleToDoneList(ModuleSymbol* module) {
 *                                                                             *
 ************************************** | *************************************/
 
+
+ModuleSymbol* parseIncludedSubmodule(const char* name) {
+  // save parser global variables to restore after parsing the submodule
+  BlockStmt*  s_yyblock = yyblock;
+  const char* s_yyfilename = yyfilename;
+  int         s_yystarlineno = yystartlineno;
+  ModTag      s_currentModuleType = currentModuleType;
+  const char* s_currentModuleName = currentModuleName;
+  int         s_chplLineno = chplLineno;
+  bool        s_chplParseString = chplParseString;
+  const char* s_chplParseStringMsg = chplParseStringMsg;
+  bool        s_currentFileNamedOnCommandLine = currentFileNamedOnCommandLine;
+
+  std::string curPath = yyfilename;
+
+  // compute the path of the file to include
+  size_t lastDot = curPath.rfind(".");
+  INT_ASSERT(lastDot < curPath.size());
+  std::string noDot = curPath.substr(0, lastDot);
+  std::string includeFile = noDot + "/" + name + ".chpl";
+
+  const char* modNameFromFile = filenameToModulename(curPath.c_str());
+  if (0 != strcmp(modNameFromFile, currentModuleName))
+    USR_FATAL("Cannot include module from module with different file name");
```

Comment:
"filename"?

Reply:
fixed

Reply:
I think there may have been other instances of this as well that I didn't flag individually (you may have fixed those as well, I haven't checked).

Reply:
(unresolving until I know one of us has as a placeholder)

Reply:
Yes there were two and I fixed both.